packages/myst-roles/src/si.ts

```diff
@@ -10,11 +10,12 @@ export const siRole: RoleSpec = {
   },
   run(data: RoleData): GenericNode[] {
     const value = data.body as string;
-    const match = value.match(/([0-9]+)\s?<([\\a-zA-Z\s]+)>/);
+    const match = value.match(/(-?\d+(,\d+)*(\.\d+(e\d+)?)?)\s?<([\\a-zA-Z\s]+)>/);
     if (!match) {
       return [{ type: 'si', error: true, value }];
     }
-    const [, number, commands] = match;
+    const number = match[1];
+    const commands = match[match.length-1];
```

Comment:
👍 This makes sense, necessary now that there are a bunch more groups in the regex.